src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -554,6 +554,11 @@ private BoundPattern BindDeconstructionPattern(DeconstructionPatternSyntax node,
                 BoundExpression deconstruct = MakeDeconstructInvocationExpression(
                     node.SubPatterns.Count, inputPlaceholder, node, diagnostics, outPlaceholders: out ImmutableArray<BoundDeconstructValuePlaceholder> outPlaceholders);
                 deconstructMethod = deconstruct.ExpressionSymbol as MethodSymbol;
+                if (deconstructMethod == null)
```

Comment:
`deconstructMethod` is a `MethodSymbol`. Use `is null` or `(object)`.

Reply:
Below you check `outPlaceholders.IsDefaultOrEmpty` to see if the deconstruction is an error, while here you check if the method is `null`. It seems like this code could be simplified significantly by settling on a single check and reusing it, including where we check `isError` below.

Reply:
That is a test for a particular subelement of the deconstruction being an error, not the whole thing. Checking that the method is null is one way the whole thing can be in error.

---
In reply to: [184870737](https://github.com/dotnet/roslyn/pull/26481#discussion_r184870737) [](ancestors = 184870737)